sqlglot/dialects/tsql.py

```diff
@@ -263,6 +263,10 @@ class Tokenizer(tokens.Tokenizer):
             "XML": TokenType.XML,
         }
 
+        # TSQL allows @ to appear as a variable/identifier prefix
```

Comment:
well if you do that, doesn't that mean you should delete table_prefix_tokens?

Reply:
Good catch, I can `pop` the `HASH` token too and get rid of it completely.

Reply:
Done